plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/UpdateNuageVspDeviceCommand.java

```diff
@@ -0,0 +1,60 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package com.cloud.agent.api;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class UpdateNuageVspDeviceCommand extends Command {
+
+    private final Map<String, String> _parametersToBeUpdated;
+
+    public UpdateNuageVspDeviceCommand(Map<String, String> parametersToBeUpdated) {
+        super();
+        this._parametersToBeUpdated = parametersToBeUpdated;
```

Comment:
To prevent the impact of side-effects from changes to the ``parametersToBeUpdated`` in post construction behavior, copy the ``Map`` and wrap it in an unmodifiable as follows:
```
this._parametersToBeUpdated = Collections.unmodifiableMap(new HashMap<String, String>(parametersToBeUpdated));
```
This change will allow the return of the parameter in ``getParametersToBeUpdated`` without modification.

An alternative, cleaner approach (which I prefer, but not commonly used in the CloudStack codebase) is to use Guava's ``ImmutableMap``.